packages/monitor-v2/test/OptimisticGovernorMonitor.ts

```diff
@@ -448,4 +496,361 @@ describe("OptimisticGovernorMonitor", function () {
+
+    // Create the proposal with multiple transactions.
+    const transactions = [
+      { to: bondToken.address, operation, value: 0, data: txnData1.data },
+      { to: bondToken.address, operation, value: 0, data: txnData2.data },
+    ];
+
+    const explanation = toUtf8Bytes("These transactions were approved by majority vote on Snapshot.");
+
+    const proposeTx = await ogModuleProxy.connect(proposer).proposeTransactions(transactions, explanation);
+
+    const proposeBlockNumber = await getBlockNumberFromTx(proposeTx);
+
+    const transactionProposedEvent = (
+      await ogModuleProxy.queryFilter(
+        ogModuleProxy.filters.TransactionsProposed(),
+        proposeBlockNumber,
+        proposeBlockNumber
+      )
+    )[0];
+
+    // Move time forward to the execution time. This also requires mining new block as the bot checks challenge window
+    // based on block time.
+    await timer.setCurrentTime(transactionProposedEvent.args.challengeWindowEnds);
+    await hardhatTime.increaseTo(transactionProposedEvent.args.challengeWindowEnds);
+    let latestBlockNumber = await ethers.provider.getBlockNumber();
+
+    const spy = sinon.spy();
+    const spyLogger = createNewLogger([new SpyTransport({}, { spy: spy })]);
+    await executeProposals(
+      spyLogger,
+      await createMonitoringParams(latestBlockNumber, { ogDiscovery: true, signer: executor, supportedBonds })
+    );
+
+    // Get the ProposalExecuted events (there should be only one).
+    latestBlockNumber = await ethers.provider.getBlockNumber();
+    const proposalExecutionEvents = await ogModuleProxy.queryFilter(
+      ogModuleProxy.filters.ProposalExecuted(),
+      proposeBlockNumber,
+      latestBlockNumber
+    );
+    assert.equal(proposalExecutionEvents.length, 1);
+
+    // When calling the bot module directly there should be only one log (index 0) with the execution caught by spy.
+    assert.equal(spy.getCall(0).lastArg.at, "oSnapAutomation");
+    assert.equal(spy.getCall(0).lastArg.message, "Submitted oSnap Execution 🏁");
+    assert.equal(spyLogLevel(spy, 0), "info");
+    assert.isTrue(spyLogIncludes(spy, 0, ogModuleProxy.address));
+    assert.isTrue(spyLogIncludes(spy, 0, transactionProposedEvent.args.proposalHash));
+    assert.isTrue(spyLogIncludes(spy, 0, space));
+    assert.isTrue(spyLogIncludes(spy, 0, proposalExecutionEvents[0].transactionHash));
+    assert.equal(spy.getCall(0).lastArg.notificationPath, "optimistic-governor");
+  });
+  it("Automatic execution not run for unsupported bonds", async function () {
+    // Deploy a new OG module proxy and approve the proposal bond.
+    const space = "test.eth";
+    const { customAvatar, ogModuleProxy } = await deployOgModuleProxy({
+      space,
+      quorum: 0,
+      votingPeriod: 0,
+    });
+    await bondToken.connect(proposer).approve(ogModuleProxy.address, await ogModuleProxy.getProposalBond());
+
+    // Empty supported bond settings.
+    const supportedBonds: SupportedBonds = {};
```

Comment:
nit: might be a bit stronger to populate this with the right token but the wrong amount and the wrong token, but the right amount to vet the checks a bit more.

Reply:
fixed